### ISSUE TITLE
Possibly unbound when using `typing.TYPE_CHECKING`

### ISSUE THREAD

**Issue by NateScarlet** [microsoft/pyright#334]
**Describe the bug**
All refererence to types that imported under typing.TYPE_CHECKING block will cause `possibly unbound` problem.

**To Reproduce**
```python
import typing
if typing.TYPE_CHECKING:
    import collections

def foo(a: 'collections.OrderedDict'):
    pass
```

**Expected behavior**
No problem reported

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/21083014/68646509-56cfb280-0556-11ea-8e6e-9af0c81ab58b.png)

**VS Code extension or command-line**
VS Code extension: 1.0.84

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version.

**Comment by erictraut** [comment]
This is now fixed in 1.1.0, which I just published.

**Comment by NateScarlet** [comment]
I am still occuring this problem on 1.1.0 and 1.1.3 .
@erictraut ?
![image](https://user-images.githubusercontent.com/21083014/69402658-7c697280-0d33-11ea-964e-abec87bd5a2c.png)

**Comment by erictraut** [comment]
Ah, my bad. My fix works if you use `TYPE_CHECKING`, but not if you use `typing.TYPE_CHECKING`. I'll expand my fix so it works with both. In the meantime, you can make it work by changing your code to the following:

```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    import collections

def foo(a: 'collections.OrderedDict'):
    pass
```

**Comment by erictraut** [comment]
This is now fixed in 1.1.4, which I just published.

**Comment by mplanchard** [comment]
Howdy! Just wanted to add that the fixes don't work if you've aliased the `typing` import to something else. We typically use `t`, e.g. `import typing as t`, because we find it cleaner than either importing a ton of symbols from typing or having to type `typing.List` or whatever everywhere.

Thanks!

**Comment by erictraut** [comment]
Thanks for pointing that out. It will be addressed in the next release.
